internals/webhooks/capp_validation_utils.go

```diff
@@ -64,18 +66,33 @@ func validateDomainName(domainname string) (errs *apis.FieldError) {
 
 // validateTlsFields checks if the fields of the tls feature in the capp spec is written correctly.
 // It takes a cappv1alpha1.Capp object and returns aggregated error if the any of the validations falied.
-func validateTlsFields(capp cappv1alpha1.Capp) (errs *apis.FieldError) {
+func validateTlsFields(capp cappv1alpha1.Capp, client client.Client, ctx context.Context) (errs *apis.FieldError) {
 	if capp.Spec.RouteSpec.TlsEnabled && capp.Spec.RouteSpec.TlsSecret == "" {
 		errs = errs.Also(apis.ErrGeneric(
 			"it's forbidden to set '.spec.routeSpec.tlsEnabled' to 'true' without specifying a secret name in the '.spec.routeSpec.tlsSecret' field"))
 	} else if !capp.Spec.RouteSpec.TlsEnabled && capp.Spec.RouteSpec.TlsSecret != "" {
 		errs = errs.Also(apis.ErrGeneric(
 			"it's forbidden to set '.spec.routeSpec.tlsEnabled' to 'false' and specifying a secret name in the '.spec.routeSpec.tlsSecret' field"))
+	} else if capp.Spec.RouteSpec.TlsEnabled && capp.Spec.RouteSpec.TlsSecret != "" && capp.DeletionTimestamp.IsZero() {
+		tlsSecret, err := getTlsSecret(capp.Spec.RouteSpec.TlsSecret, capp.Namespace, client, ctx)
+		if err != nil {
+			errs = errs.Also(apis.ErrGeneric(fmt.Sprintf(
+				"it's forbidden to set '.spec.routeSpec.tlsSecret' without creating a matching tlsSecret. %q does not exist", capp.Spec.RouteSpec.TlsSecret)))
+		} else if tlsSecret.Type != v1.SecretTypeTLS {
+			errs = errs.Also(apis.ErrGeneric(
+				fmt.Sprintf("it's forbidden to set '.spec.routeSpec.tlsSecret' without creating a matching tlsSecret. The secret type %q must be %q", tlsSecret.Name, v1.SecretTypeTLS)))
+		}
```

Comment:
Gotta add a comment here to explain why you check the `DeletionTimestamp` since it's not trivial